educational-app/src/context/ProgressContext.test.tsx

```
@@ -0,0 +1,111 @@
+import { render, screen, act, fireEvent } from '@testing-library/react'
+import { ProgressProvider, useProgress } from './ProgressContext'
+import { expect, it, describe, beforeEach, vi } from 'vitest'
+
+const TestComponent = () => {
+  const { completedSubTopics, toggleSubTopic, isSubTopicCompleted, getProgressForTopic } = useProgress()
+  return (
+    <div>
+      <div data-testid="completed-count">{completedSubTopics.length}</div>
+      <div data-testid="is-completed">{isSubTopicCompleted('test-slug') ? 'yes' : 'no'}</div>
+      <div data-testid="progress">{getProgressForTopic(['test-slug', 'other'])}</div>
+      <button onClick={() => toggleSubTopic('test-slug')}>Toggle</button>
```

Biome (2.4.4)

[error] Provide an explicit type prop for the button element.

(lint/a11y/useButtonType)